File: msg-sentiment/src/main/java/dev/number6/sentiment/dagger/ChannelMessagesHandlerModule.java
package dev.number6.sentiment.dagger;

import dev.number6.comprehend.results.PresentableSentimentResults;
import dev.number6.message.ChannelMessagesComprehensionHandler;
import dev.number6.message.ChannelMessagesHandler;
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction;
import dev.number6.message.ComprehensionResultsConsumer;
import dagger.Module;
import dagger.Provides;

@Module
public class ChannelMessagesHandlerModule {

    @Provides
    public ChannelMessagesHandler providesMessageComprehension(ChannelMessagesToComprehensionResultsFunction<PresentableSentimentResults> channelMessagesToComprehensionResultsFunction,
                                                               ComprehensionResultsConsumer<PresentableSentimentResults> comprehensionResultsConsumer) {
        return new ChannelMessagesComprehensionHandler<>(channelMessagesToComprehensionResultsFunction, comprehensionResultsConsumer);
    }
}

File: msg-sentiment/src/main/kotlin/dev/number6/sentiment/dagger/ChannelMessagesHandlerModule.kt
package dev.number6.sentiment.dagger

import dagger.Module
import dagger.Provides
import dev.number6.comprehend.results.PresentableSentimentResults
import dev.number6.message.ChannelMessagesComprehensionHandler
import dev.number6.message.ChannelMessagesHandler
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction
import dev.number6.message.ComprehensionResultsConsumer

@Module
class ChannelMessagesHandlerModule {
    @Provides
    fun providesMessageComprehension(channelMessagesToComprehensionResultsFunction: ChannelMessagesToComprehensionResultsFunction<PresentableSentimentResults>,
                                     comprehensionResultsConsumer: ComprehensionResultsConsumer<PresentableSentimentResults>): ChannelMessagesHandler {
        return ChannelMessagesComprehensionHandler(channelMessagesToComprehensionResultsFunction, comprehensionResultsConsumer)
    }
}